src/coordination/include/coordination/coordinator_instance.hpp

```diff
@@ -130,25 +138,24 @@ class CoordinatorInstance {
 
   void DemoteFailCallback(std::string_view repl_instance_name);
 
-  void ForceResetCluster();
+  auto ForceResetCluster_() -> ForceResetClusterStateStatus;
 
   auto GetBecomeLeaderCallback() -> std::function<void()>;
   auto GetBecomeFollowerCallback() -> std::function<void()>;
 
   HealthCheckClientCallback client_succ_cb_, client_fail_cb_;
   std::atomic<bool> is_leader_ready_{false};
-
+  std::atomic<bool> is_shutting_down_{false};
   // NOTE: Must be std::list because we rely on pointer stability.
   // TODO(antoniofilipovic) do we still rely on pointer stability
   std::list<ReplicationInstanceConnector> repl_instances_;
   mutable utils::ResourceLock coord_instance_lock_{};
 
-  // Thread pool needs to be constructed before raft state as raft state can call thread pool
-  utils::ThreadPool thread_pool_{1};
-
-  // This needs to be constructed last because raft state may call
-  // setting up leader instance
   std::unique_ptr<RaftState> raft_state_;
```

Comment:
Does this condition still apply or? When does thread pool need to be constructed?

Reply:
Explained on call